docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
 # Developer Guide
 
-## Acknowledgements
+## Design
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{To Be Added}
 
-## Design & implementation
+---
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Implementation
+
+### Add feature
 
+The add feature is implemented using a `AddCommand` class extending from `Command` Class. 
```

Comment:
Good job explaining how AddCommand works. It is well detailed and not too precise.